ring.go

```diff
@@ -58,3 +58,15 @@ func (r *ring) avg() int {
 	}
 	return r.sum / len(r.data)
 }
+
+func (r *ring) sequentialData() []int {
+	index := r.idx
+	slice := make([]int, r.maxLen)
+	if r.idx == 0 || len(r.data) < r.maxLen {
```

Comment:
```suggestion
	if r.idx == 0 {
```
len condition is useless here, it will always be true when index == 0.

Reply:
also, use index instead of r.idx should be better.

Reply:
`len(r.data) < r.maxLen` It is useful when the length of ring has not reached maxlen.